File: Java/Step08_Extends/src/test/auto/Car.java
package test.auto;


public class Car {

  protected Engine engine;
  
  public Car(Engine engine) {
    this.engine = engine;
  }
  
  public void drive() {
    if(this.engine == null) {
      System.out.println("no engine");
    }
    else {
      System.out.println(engine.e_name + " go go go");
    }
  }
}

File: KotlinStudy/KotlinStudy_b/bin/main/com/example/kotlin_study/mypac/Car.kt
package com.example.kotlin_study.mypac

class Car {

    var name:String? = null

    constructor() {
        println("$name constructor()")
    }
    constructor(name:String){
        this.name = name
        println("${this.name} constructor()")
    }

    fun drive() {
        println("$name is on driving")
    }
}